spring-ai-client-chat/src/main/java/org/springframework/ai/chat/client/advisor/MessageChatMemoryAdvisor.java

```diff
@@ -109,18 +131,69 @@ private void after(ChatClientResponse chatClientResponse) {
 				.map(g -> (Message) g.getOutput())
 				.toList();
 		}
-		this.getChatMemoryStore().add(this.doGetConversationId(chatClientResponse.context()), assistantMessages);
+		this.chatMemory.add(this.doGetConversationId(chatClientResponse.context()), assistantMessages);
+		return chatClientResponse;
+	}
+
+	public static Builder builder(ChatMemory chatMemory) {
+		return new Builder(chatMemory);
 	}
 
-	public static class Builder extends AbstractChatMemoryAdvisor.AbstractBuilder<ChatMemory> {
+	public static class Builder {
+
+		private String conversationId = ChatMemory.DEFAULT_CONVERSATION_ID;
+
+		private int order = Advisor.DEFAULT_CHAT_MEMORY_PRECEDENCE_ORDER;
+
+		private Scheduler scheduler;
+
+		private ChatMemory chatMemory;
 
 		protected Builder(ChatMemory chatMemory) {
```

Comment:
private instead of protected

```suggestion
		private Builder(ChatMemory chatMemory) {
```